packages/edit-site/src/components/editor-canvas-container/index.js

```diff
@@ -45,15 +45,6 @@ function getEditorCanvasContainerTitle( view ) {
 	}
 }
 
-// Creates a private slot fill.
-const { createPrivateSlotFill } = unlock( componentsPrivateApis );
-const SLOT_FILL_NAME = 'EditSiteEditorCanvasContainerSlot';
-const {
-	privateKey,
-	Slot: EditorCanvasContainerSlot,
-	Fill: EditorCanvasContainerFill,
-} = createPrivateSlotFill( SLOT_FILL_NAME );
```

Comment:
The creation and the rendering of the slot has moved to the editor package. So any editor (post or site editor) can decide to actually update the "content" are using a fill. (like we do for style book and style revisions). There are probably better ways to do implement this behavior but I didn't want to change too much here.

Reply:
> There are probably better ways to do implement this behavior but I didn't want to change too much here.

What do you think? Do you mean in terms of restricting the behavior to the post editor or finding a different way to swap editor content in and out?

Reply:
There are a few thoughts that I have here:

 - I don't like that we're switching the canvas using a Slot, I'd prefer a children prop or something maybe.
 - I wonder if we're actually switching canvas here, I wonder if stylebook or style revisions are actually the same "editor", maybe it's the whole thing that should be swapped (so no need for Slot or anything). This is clearer for things like:
 
  - Why I have an inserter and the different block tools for stylebook and style revisions (we have a dedicated prop to remove these but the prop shouldn't exist in the first place) => in other words the header is completely different 
  - The "document" sidebar or the "block" sidebar don't make sense anymore. In these modes. We're focused on "global styles". => in other words, the left sidebar is completely different
  - There's no need for list view or inserter => right sidebar is completely different.

---
This is to say, in my mind, there are two valid paths forward:

 - Build a completely different UI (the whole thing, not just the canvas) for these modes. (Could share some very small things like save button and undo/redo)
 - Continue with the "editor" but instead of hacking things around like we're doing now, switch the "active entity" (currentPostType, and currentPostId) by passing a different "post" object to EditorProvider, and that post would be the "global styles" post. In which case, the "editor" would know that we're editing the global styles and would switch the UI accordingly.

Both of these can be valid, right now, I'm thinking the former is probably better but it's a very loosely held opinion.

Reply:
Thanks a lot for the thorough feedback!

> Build a completely different UI (the whole thing, not just the canvas) for these modes. (Could share some very small things like save button and undo/redo)

I could see this as an alternative, particularly if we could isolate it to the site editor. 

I guess we have access to context/edited post in the site editor [Editor](https://github.com/WordPress/gutenberg/blob/fcedde83fc5f29efb55f76058605570a0e633c60/packages/edit-site/src/components/editor/index.js) component.

Worth an experiment. 

I'll create an issue and put it on the backlog. 

Thank you!

Reply:
The important feature, at least for global style revisions, was to have a "separate" editor view that could be edited independently of the main editor canvas.

So changes could be made to styles etc in the alt. canvas view, and when it closes, the user is returned to the main editor canvas in whatever state they left it.